src/lib/rules/no-insecure-cookies/no-insecure-cookies.ts

```diff
@@ -0,0 +1,314 @@
+                }
+
+                if (parsedSetCookie.domain) {
+                    errors.push(hasDomainHostPrefixError);
+                }
+            }
+
+            return errors;
+        };
+
+        // Validate `Secure` and `HttpOnly` attributes.
+        const validateSecurityAttributes = (parsedSetCookie: ParsedSetCookie): ValidationErrors => {
+            const cookieName: string = parsedSetCookie.name;
+            const resource: string = parsedSetCookie.resource;
+            const errors: ValidationErrors = [];
+
+            const hasSecureHttpError: string = `Insecure sites (${resource}) can't set cookies with the 'secure' directive.`;
+            const noSecureError: string = `'${headerName}' header to set '${cookieName}' doesn't have the 'secure' directive.`;
+            const noHttpOnlyError: string = `'${headerName}' header to set '${cookieName}' doesn't have the 'httponly' directive.`;
+
+            // Check against `Secure` directive if sites are insecure.
+            if (!isHTTPS(resource) && parsedSetCookie.secure) {
+                errors.push(hasSecureHttpError);
+
+                return errors;
+            }
+
+            // Check for `Secure` directive if sites are secure.
+            if (!parsedSetCookie.secure) {
+                errors.push(noSecureError);
+            }
+
+            // Check for `httpOnly` directive.
+            if (!parsedSetCookie.httponly) {
+                errors.push(noHttpOnlyError);
+            }
+
+            return errors;
+        };
+
+        // Validate `Expire` date format.
+        const validateExpireDate = (parsedSetCookie: ParsedSetCookie): ValidationErrors => {
+            const cookieName: string = parsedSetCookie.name;
+            const errors: ValidationErrors = [];
+
+            if (!parsedSetCookie.expires) {
+                return errors;
+            }
+
+            // Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
+            const utcTimeString: string = new Date(parsedSetCookie.expires).toUTCString();
+            const invalidDateError: string = `Invalid date in 'expires' value of the '${headerName}' header to set '${cookieName}'.`;
+            const invalidDateFormatError: string = `Invalid date format in 'expires' value of the '${headerName}' header to set '${cookieName}'. The recommended format is: ${utcTimeString}`;
+
+            if (utcTimeString === 'Invalid Date') {
+                errors.push(invalidDateError);
+
+                return errors;
+            }
+
+            if (normalizeString(utcTimeString) !== parsedSetCookie.expires) {
+                errors.push(invalidDateFormatError);
+            }
+
+            return errors;
+        };
+
+        // Validate browser compatibility of `Max-Age` .
+        const validateMaxAgeCompatibility = (parsedSetCookie: ParsedSetCookie): ValidationErrors => {
+            const cookieName: string = parsedSetCookie.name;
```

Comment:
@molant What should happen if the browser is different than IE6-IE8, like chrome, but ONLY `max-age` exists? According to the second scenario in the list ('only browsers different than IE6-IE8'), it should pass (If `max-age` exists: OK); But according to the third scenario ('All browsers') in the list,it should throw an error/warning because there should be a `expires`?